api_products.md

```diff
@@ -11,25 +11,214 @@ information about api access.
+| cm2          | Square centimeter                                                                                              |
+| mm           | Millimeter                                                                                                     |
+| cm           | Centimeter                                                                                                     |
+| m            | Meter                                                                                                          |
+| kg           | Kilogram                                                                                                       |
+| g            | Gram                                                                                                           |
+| t            | Tonne                                                                                                          |
+| piece        | The encoded number should be interpreted as number of contained units, i.e. number of bread rolls in a package |
+| price        | The encoded number should be interpreted as total price, i.e. price of a package of different sausages         |
+
+
+### Code Template
+
+Code templates are a way to extract information embeded into a scanned
+code. For example a instore EAN-13 has the format `2xxxxxCeeeeeC`
+where the five `x` encode an identfier for the product, the five `e`
+encode the the information and the `C` are the checksums defined by
+the standard. With this and the additional inforations contained in
+the product, it is possible to calculate the price.
+
+For example: Given the scanned code `2123455005005`. Using the
+template it is possible to extract the identifier `12345` and the
+embeded information `00500`. A lookup in products information might
+yield, that this code belongs to the product `Banana` wich has a price
+of 2.00 €, `g` as encoding unit and `kg` as reference unit. With this
+information it is possible to calculate the price: `00500` → 500g →
+0,5 kg × 2.00 € → 1 €.
+
+| Template             | Description                                                      |
+|----------------------|------------------------------------------------------------------|
+| `default`            | The default template, matches anything                           |
+| `ean14_code128`      | An EAN-14 embedded in a Code128 with Application Identifier "01" |
+| `ean13_instore`      | An in-store EAN-13 with embedded data, but no internal checksum  |
+| `ean13_instore_chk`  | An in-store EAN-13 with embedded data and internal checksum      |
+| `edeka_discount`     | Price reduction for arbitrary products                           |
+| `german_print`       | German newspapers/magazines with embedded price                  |
+
+### Deposits
+
+All products might reference a deposit, which is automatically added
+to shopping cart if a user buys the products. These products can not
+be bought without buying the product.
```

Comment:
Alternative, klare Formulierung:
"Deposit products can not be bought on their own."